packages/react/src/components/StructuredList/StructuredList.js

```diff
@@ -182,7 +182,13 @@ export function StructuredListRow(props) {
       {...other}
       role="row"
       className={classes}
-      onClick={() => setSelectedRow(id)}
+      onClick={(event) => {
+        setSelectedRow(id);
+        // Avoid that the onClick gets called when arrow down
+        if (!event.target.className.includes('cds--structured-list-input')) {
+          onClick && onClick();
+        }
```

Comment:
@guidari Could you describe more of what this is intending to avoid?

Reply:
When the user presses the ArrowDown key, the `onClick` method is called because the input element is considered to be "clicked." This code allows us to split the `onClick` and `onKeyDown` events into different behaviors.

Reply:
Ah interesting, I forgot arrow navigation selected the input. Relying on the classname of a different component feels like we're not maintaining a clear separation of concerns between these two components. I know they're related, but it still feels funky.

Would it be okay if we just passed this all the way through to the consumer no matter if it's on the input or not and then they could do this check themselves instead if they need to?

Reply:
@tay1orjones You mean that we should let to the consumer do this check instead of us? Sure, we can do that.

I just feel confunsing the `onClick` being triggred when the user arrow down, maybe the user would think the same. But if we remove the checker should we point it out, or just let it be?

Reply:
@guidari Wow, sorry for such a delayed response! Yes I think it's okay to leave it as is and let folks filter out the events they don't need

Reply:
Cool! I just removed @tay1orjones